jobs/pywikibot/scripts/updatestatistics.py

```diff
@@ -151,10 +166,20 @@ def main():
     # Authenticate and construct service.
     service = get_service('analytics', 'v3', scope, key_file_env)
     profile = get_first_profile_id(service)
+
     popular_pages = get_popular_pages(service, profile)
+    popular_pages = popular_pages[:REQD_TEMPLATE_LEN]
+
     update_list_of_pages('Template:Popular_pages', popular_pages)
+
     trending_pages = get_trending_pages(service, profile)
-    update_list_of_pages('Template:Trending_pages', trending_pages)
+    trending_pages_deduped = deduplicate_lists( \
+                                trending_pages, \
+                                popular_pages \
```

Comment:
This should be `popular_pages[:REQD_TEMPLATE_LEN]`, otherwise we're potentially removing trending items that aren't going to show in popular anyway.

Reply:
This truncation is done at L171. 

[`popular_pages = popular_pages[:REQD_TEMPLATE_LEN]`](https://github.com/metakgp/metakgp-wiki/blob/65ad097625f636d85961ca187dc253ccb78f3527/jobs/pywikibot/scripts/updatestatistics.py#L171)

Reply:
Ah yes, dammit mutability!